src/lib/Web3Function.ts

```diff
@@ -121,7 +131,10 @@ export class Web3Function {
       },
     };
 
-    const result = await this._onRun(context);
+    const result = ctxData.log
+      ? await (this._onRun as eventRunHandler)({ ...context, log: ctxData.log })
+      : await (this._onRun as baseRunHandler)(context);
+
```

Comment:
I think you could just add: 
```
      log: ctxData. log,
```
on the `context` creation, and keep:
```
    const result = await this._onRun(context);
```

Reply:
For DX, I wanted to enforce log is defined for event triggering functions

Reply:
I don't understand what is the difference in DX between the suggestion:
```
const context = {
  ...
  log: ctxData.log
}
const result = await this._onRun(context);
```
And your implementation?
as your condition is on `ctxData.log`